oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,499 @@
+locally checked out ``edx-platform``, and the code of the devstack itself (i.e, Tutor) is also locally modifiable, you
+must:
+
+1. Create a self-contained workspace for the devstack, check out Tutor and install requirements:
+
+   .. code-block::
+
+      export WORKSPACE=~/workspace
+      mkdir -p $WORKSPACE
+      cd $WORKSPACE
+      git clone git@github.com:overhangio/tutor.git
+      cd tutor
+      pip install -e .
+
+2. Follow Tutor's instructions to prepare the default development environment, setting ``TUTOR_ROOT`` so configuration
+   is stored independently, and making sure to answer "no" when asked if this is a production environment:
+
+   .. code-block::
+
+      export TUTOR_ROOT=${WORKSPACE}/tutor_root
+      tutor local quickstart
+      tutor local stop
+      tutor images build --no-cache openedx-dev
+
+3. Because the master branch of ``edx-platform`` is not supported at the time of writing, one must clone Open edX at the
+   latest supported release *tag*:
+
+   .. code-block::
+
+      cd $WORKSPACE
+      git clone -b open-release/koa.1 git@github.com:edx/edx-platform.git
+
+   .. note::
+
+      Note that branches based off of ``master``, or even ``koa.master``, will not work in production.  As noted
+      `in the documentation <https://docs.tutor.overhang.io/configuration.html#running-a-fork-of-edx-platform>`__, "Tutor
+      will attempt to apply security and bug fix patches that might already be included", and possibly fail.
+
+4. Create a docker-compose override file for the development environment:
+
+   .. code-block::
+
+      vim ${TUTOR_ROOT}/env/dev/docker-compose.override.yml
+
+   And configure it so that your local checkout of ``edx-platform`` is bind-mounted in the correct directory by all
+   relevant containers:
+
+   .. code-block::
+
+      version: "3.7"
+      services:
+        lms:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        cms:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        lms-worker:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        cms-worker:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+
+5. Prepare the ``edx-platform`` repository inside the LMS container:
+
+   .. code-block::
+
+      tutor dev run lms bash
+      pip install -r requirements/edx/development.txt
+      npm install
+      openedx-assets build --env=dev
+      exit
+
+6. You'll need to init the dev environment.  (This is taken care of by ``tutor local quickstart`` for the local
+   production environment, and even though the same databases are used for the development environment, apparently the
+   ``GRANT`` command needs to be run again when MySQL runs on a different container.)
+
+   .. code-block::
+
+      tutor dev init
+
+7. And finally, to start the LMS, CMS, and workers in the background:
+
+   .. code-block::
+
+      tutor dev start -d
```

Comment:
This process is fairly complex -- that's because it is trying to replicate the way things work on the Devstack. And the Devstack-way-of-doing-thing® is not the Tutor-way-of-doing-things® :stuck_out_tongue: 

Now that [automatic bind-mounts](https://docs.tutor.overhang.io/dev.html#bind-mount-from-the-volumes-directory) exist, when I need to develop on edx-platform I typically run the following commands:

```
tutor dev bindmount lms /openedx/edx-platform
tutor dev runserver --volume=/openedx/edx-platform lms
```

Then I edit the files from `$(tutor local printroot)/volumes/edx-platform`.

Reply:
> the Devstack-way-of-doing-thing® is not the Tutor-way-of-doing-things®

@regisb, I agree completely.  And this isn't really a judgement call on which is better.  It's just that I specifically set out to get as close to a devstack as I could, as an exercise in comparing them.

To be fair, if we were to try and get the devstack to behave as Tutor... It'd be much, much harder. :)